Compilation on Raspberry Pi B+ armv6l

Hi there,

I am trying to compile torch on a Raspberry Pi B+ with armv6l (on raspian) and it fails reporting the following error:

```
[  3%] Built target luaT
[  6%] Building C object lib/TH/CMakeFiles/TH.dir/THVector.c.o
In file included from /home/pi/torch/pkg/torch/lib/TH/THVector.c:2:0:
/home/pi/torch/pkg/torch/lib/TH/generic/THVectorDispatch.c: In function ‘THByteVector_vectorDispatchInit’:
/home/pi/torch/pkg/torch/lib/TH/generic/simd/simd.h:64:3: error: impossible constraint in ‘asm’
   asm volatile ( "cpuid\n\t"
   ^
/home/pi/torch/pkg/torch/lib/TH/generic/simd/simd.h:64:3: error: impossible constraint in ‘asm’
   asm volatile ( "cpuid\n\t"
   ^
lib/TH/CMakeFiles/TH.dir/build.make:330: recipe for target 'lib/TH/CMakeFiles/TH.dir/THVector.c.o' failed
make[2]: *** [lib/TH/CMakeFiles/TH.dir/THVector.c.o] Error 1
CMakeFiles/Makefile2:129: recipe for target 'lib/TH/CMakeFiles/TH.dir/all' failed
make[1]: *** [lib/TH/CMakeFiles/TH.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2

Error: Build error: Failed building.
```

More details about the raspberry below:

```
pi@raspberrypi:~/torch $ uname -a
Linux raspberrypi 4.4.11+ #888 Mon May 23 20:02:58 BST 2016 armv6l GNU/Linux
```

I have seen a similar issue on https://github.com/torch/torch7/issues/794
Is there a related fix?

--- a/lib/TH/generic/simd/simd.h
+++ b/lib/TH/generic/simd/simd.h
@@ -57,6 +57,14 @@ static inline uint32_t detectHostSIMDExtensions()
 
 #else // x86
 
+#if defined(__arm__) //ARM without NEON
+
+static inline uint32_t detectHostSIMDExtensions()
+{
+  return SIMDExtension_DEFAULT;
+}
+  
+#else
 static inline void cpuid(uint32_t *eax, uint32_t *ebx, uint32_t *ecx, uint32_t *edx)
 {
 #ifndef _MSC_VER
@@ -99,5 +107,6 @@ static inline uint32_t detectHostSIMDExtensions()
 }
 
 #endif // end x86 SIMD extension detection code
+#endif // end __arm__
 
 #endif

